pkg/builder/entity_dto_builder.go

```diff
@@ -221,12 +221,35 @@ func (eb *EntityDTOBuilder) DisplayName(displayName string) *EntityDTOBuilder {
 	return eb
 }
 
+// remove duplicate commodities from the given commodity list
+func PurifyCommodities(commDTOs []*proto.CommodityDTO) []*proto.CommodityDTO {
+	commodityMap := make(map[string]*proto.CommodityDTO)
+	for _, commodity := range commDTOs {
+		mapKey := string(*commodity.CommodityType)
+		if commodity.Key != nil {
+			mapKey = mapKey + "_" + string(*commodity.Key)
+		}
+
+		if _, found := commodityMap[mapKey]; !found {
```

Comment:
This check is unnecessary and code will be faster without this.
Line 234 alone should be sufficient. Duplicate commodities are rare and if there is a duplicate, the commodity value will be set to the last one found.

Reply:
Done